velox/expression/Expr.cpp

```diff
@@ -271,6 +271,135 @@ void Expr::releaseInputValues(EvalCtx& evalCtx) {
   inputValues_.clear();
 }
 
+namespace {
+
+void ensureErrorsSize(EvalCtx::ErrorVector& errors, int32_t size) {
+  if (size > errors.size()) {
+    int32_t oldSize = errors.size();
+    errors.resize(size);
+    for (auto i = oldSize; i < size; ++i) {
+      errors.setNull(i, true);
+    }
+  }
+}
+
+// Adds the errors from 'errors' to 'result'.
+void addErrors(
+    const EvalCtx::ErrorVector& errors,
+    EvalCtx::ErrorVector& result) {
+  ensureErrorsSize(result, errors.size());
+  bits::forEachBit(
+      errors.rawNulls(), 0, errors.size(), bits::kNotNull, [&](auto row) {
+        result.set(row, errors.valueAt(row));
+      });
+}
+
+void rethrowFirstError(const EvalCtx::ErrorVectorPtr& errors) {
```

Comment:
It would be nice to consolidate all error throwing code in one place. Currently, that's in EvalCtx.